source/library/utilities/get-pattern-mtimes.js

```diff
@@ -0,0 +1,87 @@
+import {dirname, basename, resolve} from 'path';
+import {readFile} from 'fs';
+import fs from 'q-io/fs';
+import denodeify from 'denodeify';
+import {stat} from 'fs';
+import {find} from 'lodash';
+
+import pathToId from '../resolve-utilities/path-to-id';
+const fsStat = denodeify(stat);
+const read = denodeify(readFile);
+
+async function readManifest(path) {
+	return await read(resolve(path, 'pattern.json'))
+		.then(content => JSON.parse(content.toString('utf-8')));
+}
+
+async function getPatternFilesMtime(files) {
+	const tasks = files.map(async file => {
+		return (await fsStat(file)).mtime;
+	})
+
+	return await* tasks;
+}
+
+function getLatestMtime(mtimes) {
+	const times = mtimes.map(mtime => {
+		return {stamp: mtime.getTime(), date: mtime}
+	});
+
+	const latest = times.sort((a, b) => b.stamp - a.stamp)[0];
+	return latest.date;
+}
+
+function getDependencyMtimes(pattern, patterns) {
```

Comment:
Okay for now I think, but this should belong to a global cache so all tasks and transforms can access this and do not have to do file-io all the time.